### DISCUSSION TITLE
'Bad template syntax' if comment is missing

### DISCUSSION THREAD

**Discussion by ilandikov** [SilentVoid13/Templater#790] [Help] [answered]
Hello,

first of all this is my first discussion and I'm not much of a programmer so any suggestions on how to make this report/request better are appreciated.

As some weeks are shared by 2 months, I try to add the reference for both in such cases. I have written the following code:
```
<%*
    let referencedMonth = "\[\[" + tp.date.weekday("YYYY-MM", 0) + "\]\]"
    if (tp.date.weekday("MM", 0) != tp.date.weekday("MM", 6)) {
        referencedMonth += " \[\[" + tp.date.weekday("YYYY-MM", 6) + "\]\]"
        // additionalMonth = "--"
    }
-%>
<< [[<% tp.date.now("YYYY[-Н]WW", "P-1W") %>]] | <% referencedMonth %> | [[<% tp.date.now("YYYY[-Н]WW", "P1W") %>]] >>
```

Everything works fine until I remove the comment `// additionalMonth = "--"`. In this case I have `Bad template syntax Unexpected token '%'` error thrown.

Am I doing something wrong? Any help is appreciated.

**Comment by welpdx** [comment, marked as answer]
The problem here is that you are using `"` to enclose the backslash `\`. You should use tilda `` ` `` instead. 
Normally for the square brackets, using quotes would do (e.g., `"[["`) . But since you added backslash, you should use tilda (e.g., `` `\[\[` ``. 

Honestly though, I recommend you remove the backslashes. You don't really need to escape the square brackets here. 



#### with backslash
```js
<%*
	let referencedMonth = `\[\[` + tp.date.weekday("YYYY-MM", 0) + `\]\]`
	if (tp.date.weekday("MM", 0) != tp.date.weekday("MM", 6) ) {
	referencedMonth += ` `+ `\[\[` + tp.date.weekday("YYYY-MM", 6) + `\]\]` 
	}
-%>

<< [[<% tp.date.now("YYYY[-Н]WW", "P-1W") %>]] | <% referencedMonth %> | [[<% tp.date.now("YYYY[-Н]WW", "P1W") %>]] >>
```


#### without backslash
```js
<%*
	let referencedMonth = "[[" + tp.date.weekday("YYYY-MM", 0) + "]]"
	if (tp.date.weekday("MM", 0) != tp.date.weekday("MM", 6) ) {
	referencedMonth += " "+ "[[" + tp.date.weekday("YYYY-MM", 6) +"]]"
	}
-%>

<< [[<% tp.date.now("YYYY[-Н]WW", "P-1W") %>]] | <% referencedMonth %> | [[<% tp.date.now("YYYY[-Н]WW", "P1W") %>]] >>
```

**Reply by ilandikov** [reply]
Thanks a lot for your help! This works =)

Could you please enlighten me on the difference of use of `\[\[` and "\[\[" and "[[". Or maybe there is a page that does so.

I'm also wondering if there could be a way to make this look better because writing "[[" makes the thing look like

```
referencedMonth += " " + tp.date.weekday("YYYY-MM", 6) + ""
```

eg the brackets are hidden and `" + tp.date.weekday("YYYY-MM", 6) + "` part is being processed as a [[]] link.

**Reply by welpdx** [reply]
#### about backslash
Obsidian used to not recognize `\[\[texthere\]\]` as a valid internal link. But when I was testing it yesterday, I realized it now recognizes it as an internal link. So idk the difference really. 
I guess what I am saying is that in markdown and in other languages, the backslash acts as an [escape character](https://en.wikipedia.org/wiki/Escape_character). Redundant in terms of making internal links.

#### on how it looks
Yea, Obsidian isn't really made for coding. Why it looks like that is Obsidian's new live view, the WYSIWYG sorta view. I used Obsidian since the beginning so I still enjoy the legacy viewer (which doesn't have that behaviour). Alternatively, you can enclose your code in tags (i.e., 
```` 
```js 
stuff
```
````
). And then just remove it after you are done. Alternatively X2, you can use a third party software to do your programming and just save it as a .md. For example, you can change language for .md in n++ to have syntax highlighting for js. Then just open the .md file via n++ and then save it after you are done. (see [User Defined Languages](https://npp-user-manual.org/docs/user-defined-language-system/) and [superuser ans](https://superuser.com/a/40878/747891) on how to do that.)

**Reply by AB1908** [reply]
Escaping behavior just changed with the new preview release I think.

**Reply by ilandikov** [reply]
@welpdx thanks a lot for the explanation!

**Reply by welpdx** [reply]
@ilandikov 
You're welcome!
